digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/csvDataExporter/AssetReportCsvExporter.scala

```diff
@@ -0,0 +1,264 @@
+            val currentAssetType = linearAssetsGroupedBy(assetType)
+
+            val (operatorUser, operatorDate) = extractUserAndDateTime(currentAssetType, operatorUsersUsername)
+            val (municipalityUser, municipalityDate) = extractUserAndDateTime(currentAssetType, municipalityUsers)
+
+            val firstElem = currentAssetType.head
+
+            Some( ExportAssetReport(municipalitiesInfo.filter(_.id == municipality).head.name, firstElem.assetType, currentAssetType.size,
+                              firstElem.assetNameFI, firstElem.assetGeometryType, operatorUser, operatorDate, municipalityUser, municipalityDate) )
+          }
+          else
+            None
+        }
+
+      }
+      else
+        None
+    }
+  }
+
+
+  def extractDateToString(dateTime: Option[DateTime]): String = {
+    dateTime match {
+      case Some(date) =>
+        DateParser.dateToString(date, DateParser.DatePropertyFormat)
+      case _ => ""
+    }
+  }
+
+
+  def extractRecords( exportAssetReports: List[ExportAssetReport], withTransaction: Boolean = true): String = {
+
+    def getTotalNewLawTrafficSigns(municipalityName: String): Int = {
+      if(withTransaction) {
+        withDynSession {
+          val municipalityInfo = municipalityDao.getMunicipalityIdByName(municipalityName)
+          assetReporterDAO.getTotalTrafficSignNewLaw(municipalityInfo.head.id)
+        }
+      }
+      else{
+        val municipalityInfo = municipalityDao.getMunicipalityIdByName(municipalityName)
+        assetReporterDAO.getTotalTrafficSignNewLaw(municipalityInfo.head.id)
+      }
+    }
+
+
+    val output = if (exportAssetReports.isEmpty) {
+        List()
+      }
+      else {
+        val allInfoGroupedByMunicipality = exportAssetReports.groupBy(_.municipalityName)
+
+        val result = allInfoGroupedByMunicipality.map { case (municipality, records) =>
+          val line = municipality + ";\r\n"
+
+          val rows = records.map { elem =>
+                val (totalPoints, isLinear) = if (elem.assetGeomType.trim == "linear") ("", "Kyllä")
+                                              else (elem.totalAssets, "")
+
+                val operatorDate = extractDateToString(elem.operatorModifiedDate)
+                val municipalityDate = extractDateToString(elem.municipalityModifiedDate)
+
+                val baseRow = s";${elem.assetNameFI};$totalPoints;$isLinear;$operatorDate;$municipalityDate;${elem.operatorUser};${elem.municipalityUser}"
+
+                if (elem.assetTypeId == TrafficSigns.typeId) {
+                  val totalNewLawTrafficSigns = getTotalNewLawTrafficSigns(municipality)
+                  baseRow.concat(s";$totalNewLawTrafficSigns")
+                }
+                else
+                  baseRow
+                }
+
+          line + rows.mkString("\r\n")
+        }
+
+      result.toList
+    }
+
+    output.mkString("\r\n")
+  }
+
+
+  def decodeAssetsToProcess(assetTypesList: List[Int]): List[Int] = {
```

Comment:
I would prefer to return a Set[Int] so no asset type Id is equal, just in case. But this is fine too